src/theory/arith/arith_msum.cpp

```diff
@@ -81,7 +81,8 @@ bool ArithMSum::getMonomialSum(Node n, std::map<Node, Node>& msum)
 
 bool ArithMSum::getMonomialSumLit(Node lit, std::map<Node, Node>& msum)
 {
-  if (lit.getKind() == GEQ || lit.getKind() == EQUAL)
+  if (lit.getKind() == GEQ
+      || (lit.getKind() == EQUAL && lit[0].getType().isRealOrInt()))
```

Comment:
This guards against being called with `lit[0]` being Boolean, right? Shouldn't this rather be checked by the caller?

Reply:
Well its already guarding for the kind of literal here (GEQ, EQUAL), so this check should go here.